beetsplug/edit.py

```diff
@@ -262,10 +291,20 @@ def edit_objects(self, objs, fields):
                         return False
 
                 # Show the changes.
+                # If the objects are not on the DB yet, we need a copy of their
+                # original state for show_model_changes.
+                if all(not obj.id for obj in objs):
```

Comment:
In the end, using a `dict(obj)` didn't work either - the main issue is that we need to use something that is understood by `ui.show_model_changes`, and I'm not really sure we can use anything other than a DBModel without modifying that function (or providing a similar alternative that works on plain dicts, etc).

In any case, I have tidied up a bit the code by always generating `objs_old` (with a `deepcopy` if the object is not in DB, `None` otherwise) and removing the `if`s by iterating both lists at the same time - hope it makes sense.